google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java

```diff
@@ -0,0 +1,670 @@
+  private static final Key KEY4 = Key.builder(KEY2).name("newName1").namespace(NAMESPACE).build();
+  private static final Key KEY5 = Key.builder(KEY2).name("newName2").namespace(NAMESPACE).build();
+  private static final KeyValue KEY_VALUE = KeyValue.of(KEY1);
+  private static final ListValue LIST_VALUE1 = ListValue.builder()
+      .addValue(NULL_VALUE)
+      .addValue(STR_VALUE, BOOL_VALUE)
+      .build();
+  private static final ListValue LIST_VALUE2 = ListValue.of(Collections.singletonList(KEY_VALUE));
+  private static final ListValue EMPTY_LIST_VALUE = ListValue.of(Collections.<Value<?>>emptyList());
+  private static final DateTimeValue DATE_TIME_VALUE = new DateTimeValue(DateTime.now());
+  private static final LatLngValue LAT_LNG_VALUE =
+      new LatLngValue(LatLng.of(37.422035, -122.084124));
+  private static final FullEntity<IncompleteKey> PARTIAL_ENTITY1 =
+      FullEntity.builder(INCOMPLETE_KEY2).set("str", STR_VALUE).set("bool", BOOL_VALUE)
+          .set("list", LIST_VALUE1).build();
+  private static final FullEntity<IncompleteKey> PARTIAL_ENTITY2 =
+      FullEntity.builder(PARTIAL_ENTITY1).remove("str").set("bool", true)
+          .set("list", LIST_VALUE1.get()).build();
+  private static final FullEntity<IncompleteKey> PARTIAL_ENTITY3 =
+      FullEntity.builder(PARTIAL_ENTITY1).key(IncompleteKey.builder(PROJECT_ID, KIND3).build())
+          .build();
+  private static final Entity ENTITY1 = Entity.builder(KEY1)
+      .set("str", STR_VALUE)
+      .set("date", DATE_TIME_VALUE)
+      .set("latLng", LAT_LNG_VALUE)
+      .set("bool", BOOL_VALUE)
+      .set("partial1", EntityValue.of(PARTIAL_ENTITY1))
+      .set("list", LIST_VALUE2)
+      .set("emptyList", EMPTY_LIST_VALUE)
+      .build();
+  private static final Entity ENTITY2 = Entity.builder(ENTITY1).key(KEY2).remove("str")
+      .set("name", "Dan").setNull("null").set("age", 20).build();
+  private static final Entity ENTITY3 = Entity.builder(ENTITY1).key(KEY3).remove("str")
+      .set("null", NULL_VALUE).set("partial1", PARTIAL_ENTITY2).set("partial2", ENTITY2).build();
+
+  @AfterClass
+  public static void afterClass() {
+    HELPER.deleteNamespace();
+  }
+
+  @Before
+  public void setUp() {
+    DATASTORE.put(ENTITY1, ENTITY2);
+  }
+
+  @After
+  public void tearDown() {
+    DATASTORE.delete(KEY1, KEY2, KEY3);
+  }
+
+  @Test
+  public void testNewTransactionCommit() {
+    Transaction transaction = DATASTORE.newTransaction();
+    transaction.add(ENTITY3);
+    Entity entity2 = Entity.builder(ENTITY2)
+        .clear()
+        .setNull("bla")
+        .build();
+    transaction.update(entity2);
+    transaction.delete(KEY1);
+    transaction.commit();
+
+    List<Entity> list = DATASTORE.fetch(KEY1, KEY2, KEY3);
+    assertNull(list.get(0));
+    assertEquals(entity2, list.get(1));
+    assertEquals(ENTITY3, list.get(2));
+    assertEquals(3, list.size());
```

Comment:
This is there just to ensure that there are no more entities than expected. In case an entity is missing we'll fail earlier with an exception.